pkg/collector/process_test.go

```diff
@@ -144,6 +146,60 @@ func TestTCPCollectingProcess_ReceiveDataRecord(t *testing.T) {
 	assert.Equal(t, int64(1), cp.GetNumRecordsReceived())
 }
 
+// This test was added to easily measure memory usage when collecting and storing data records.
+func TestTCPCollectingProcess_ReceiveDataRecordsMemoryUsage(t *testing.T) {
+	input := getCollectorInput(tcpTransport, false, false)
+	cp, err := InitCollectingProcess(input)
+	require.NoError(t, err)
+	// Add the templates before sending data record
+	cp.addTemplate(uint32(1), uint16(256), elementsWithValueIPv4)
+
+	go cp.Start()
+
+	// wait until collector is ready
+	waitForCollectorReady(t, cp)
+
+	var conn net.Conn
+	collectorAddr := cp.GetAddress()
+	conn, err = net.Dial(collectorAddr.Network(), collectorAddr.String())
+	require.NoError(t, err)
+
+	const numRecords = 1000
+
+	ies := make([]entities.InfoElementWithValue, 0, numRecords)
```

Comment:
Apology if I didn't fully understand the test. Is there a need to collect and validate the `sourceIPv4Address ` ie values?

Reply:
I'll write a comment. If these are not collected / used after calling `runtime.ReadMemStats`, everything will be GC'd.